src/Integration/LocalServices/FileTracker.cs

```diff
@@ -58,26 +63,26 @@ public void RemoveFiles(params string[] removedFiles)
 
     public void RenameFiles(string[] beforeRenameFiles, string[] afterRenameFiles)
     {
-        threadHandling.RunOnBackgroundThread(() => NotifySlCoreFilesChangedAsync(beforeRenameFiles, afterRenameFiles)).Forget();
+        threadHandling.RunOnBackgroundThread(() => NotifySlCoreFilesChangedAsync(beforeRenameFiles, afterRenameFiles))
+            .Forget();
     }
 
     private Task NotifySlCoreFilesChangedAsync(string[] removedFiles, string[] addedFiles)
     {
         if (serviceProvider.TryGetTransientService(out IFileRpcSLCoreService fileRpcSlCoreService))
         {
-            var clientFiles = addedFiles.Select(f => new ClientFileDto(
-                new FileUri(f),
-                f.Substring(activeConfigScopeTracker.Current.RootPath.Length),
-                activeConfigScopeTracker.Current.Id,
-                null,
-                Encoding.UTF8.WebName,
-                f)).ToList();
-            
+            var rootPath = activeConfigScopeTracker.Current.RootPath;
+            var configScopeId = activeConfigScopeTracker.Current.Id;
+            var clientFiles = addedFiles.Select(fp => clientFileDtoFactory.Create(fp, configScopeId, rootPath)).ToList();
             var removedFileUris = removedFiles.Select(f => new FileUri(f)).ToList();
-                
+
             fileRpcSlCoreService.DidUpdateFileSystem(new DidUpdateFileSystemParams(
                 removedFileUris, clientFiles));
         }
+        else
+        {
+            logger.WriteLine(Strings.FileTrackerGetRpcServiceError);
```

Comment:
Use `SLCoreStrings.ServiceProviderNotInitialized` if possible, maybe with some prefix to indicate where it happened. We've not been adding the prefix in other places, but maybe we should start to.